### PR TITLE
[source-hubspot] - Handle custom objects where name collides with standard objects

### LINKED ISSUE #40679: [source-hubspot] DuplicateKey error during stream refresh when custom object has same name as standard object
### Connector Name

source-hubspot

### Connector Version

4.2.7

### What step the error happened?

Configuring a new connector

### Relevant information

Create a custom object in hubspot called "owners". Try to refresh the schema.

### Relevant log output

```shell
Internal Server Error: Duplicate key io.airbyte.protocol.models.StreamDescriptor@417723ad[name=owners,namespace=<null>,additionalProperties={}] (attempted merging values io.airbyte.protocol.models.AirbyteStream@79d017cc[name=owners,jsonSchema={"$schema":"http://json-schema.org/draft-07/schema#","type":["null","object"],"properties":{"archived":{"type":["null","boolean"]},"lastName":{"type":["null","string"]},"createdAt":{"format":"date-time","type":["null","string"]},"firstName":{"type":["null","string"]},"teams":{"type":["null","array"],"items":{"type":"object","properties":{"name":{"type":["null","string"]},"id":{"type":["null","string"]},"membership":{"type":["null","string"]}}}},"id":{"type":["null","string"]},"userId":{"type":["null","integer"]},"email":{"type":["null","string"]},"updatedAt":{"format":"date-time","type":["null","string"]}}},supportedSyncModes=[full_refresh, incremental],sourceDefinedCursor=true,defaultCursorField=[updatedAt],sourceDefinedPrimaryKey=[[id]],namespace=<null>,additionalProperties={}] and io.airbyte.protocol.models.AirbyteStream@4ad67f7c[name=owners,jsonSchema={"$schema":"http://json-schema.org/draft-07/schema#","additionalProperties":true,"type":["null","object"],"properties":{"archived":{"type":["null","boolean"]},"createdAt":{"format":"date-time","type":["null","string"]},"id":{"type":["null","string"]},"properties":{"type":["null","object"],"properties":{"hs_all_accessible_team_ids":{"type":["null","string"]},"hs_user_ids_of_all_notification_unfollowers":{"type":["null","string"]},"hs_all_team_ids":{"type":["null","string"]},"hs_read_only":{"type":["null","boolean"]},"hs_object_source_detail_1":{"type":["null","string"]},"hs_object_source_detail_3":{"type":["null","string"]},"hs_object_source_detail_2":{"type":["null","string"]},"hs_object_source_label":{"type":["null","string"]},"hubspot_team_id":{"type":["null","string"]},"hubspot_owner_id":{"type":["null","string"]},"hs_unique_creation_key":{"type":["null","string"]},"hs_object_source_id":{"type":["null","string"]},"hs_user_ids_of_all_notification_followers":{"type":["null","string"]},"hs_created_by_user_id":{"type":["null","number"]},"hs_createdate":{"format":"date-time","type":["null","string"]},"hs_was_imported":{"type":["null","boolean"]},"hs_all_owner_ids":{"type":["null","string"]},"owner_email":{"type":["null","string"]},"owner_name":{"type":["null","string"]},"hs_pinned_engagement_id":{"type":["null","number"]},"hs_user_ids_of_all_owners":{"type":["null","string"]},"hs_all_assigned_business_unit_ids":{"type":["null","string"]},"hs_lastmodifieddate":{"format":"date-time","type":["null","string"]},"hubspot_owner_assigneddate":{"format":"date-time","type":["null","string"]},"hs_object_id":{"type":["null","number"]},"hs_shared_user_ids":{"type":["null","string"]},"hs_object_source":{"type":["null","string"]},"hs_merged_object_ids":{"type":["null","string"]},"hs_shared_team_ids":{"type":["null","string"]},"hs_object_source_user_id":{"type":["null","number"]},"hs_updated_by_user_id":{"type":["null","number"]}}},"updatedAt":{"format":"date-time","type":["null","string"]}}},supportedSyncModes=[full_refresh, incremental],sourceDefinedCursor=true,defaultCursorField=[updatedAt],sourceDefinedPrimaryKey=[[id]],namespace=<null>,additionalProperties={}])
```

### Contribute

- [X] Yes, I want to contribute

### PR DESCRIPTION
**Pull request by jtruty**
## What
This handles custom object names that collide with standard object names. Currently if this is the case, stream discovery fails.
Fixes: https://github.com/airbytehq/airbyte/issues/40679

## How
We check the custom object names against a set of standard hubspot object names. If there is a collision, we use the fully qualified name for the custom object instead of the actual object name.

## Review guide
airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py

## User Impact
Stream discovery should work when this happens and allow a normal sync to work.

## Can this PR be safely reverted and rolled back?
- [X] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py [hunk 1]
```diff
@@ -220,10 +220,49 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
 
         return available_streams
 
+    # set of standard stream names to check for custom object name collisions
+    STANDARD_STREAM_NAMES = {
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py:224]: My only concern is that this is hardcoded. Although the streams won't change very frequently, it's easy to lose track of this dependency. I don't think the stream object has access to the catalog, which makes complicated to solve the problem. @airbytehq/critical-connectors  any suggestion?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py (hunk 1 above)
Comment (jtruty) [reply airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py:224]: Agreed, I'd love a more future-proof solve. I didn't see an easy way to get at the basic stream list though.

**Comment by marcosmarxm** [comment]
@airbytehq/critical-connectors can someone take a look into this contribtion?

**Comment by marcosmarxm** [comment]
@jtruty can you allow maintainers to make commits and changes to your branch?

**Pull request closed without merging**